### PR TITLE
[GSIP-164] Promote geofence modules from Community to Extension

### PR DESCRIPTION
**Pull request by afabiani**
Working on missing pieces of 

https://github.com/geoserver/geoserver/wiki/GSIP-164

- Added geofence documentation
- Increased overall code coverage

### CODE REVIEW

File: doc/en/user/source/community/geofence/configuration.rst [hunk 1]
```diff
@@ -41,3 +43,77 @@ Configure the following settings here:
...
+- Open the GeoFence admin page; you'll get to this page:
+
+   You can notice here the information that allow the GeoFence probe inside GeoServer to communicate with the GeoFence engine:
+
+   - the URL that the probe shall use to communicate with GeoFence;
+   - the name (default is `default-gs`) this instance will use to identify itself to GeoFence. This instance name should be equal to the one we set into GeoFence.
+
+- Testing connection to GeoFence.
+
+   We altready performed a connection test from GeoFence to GeoServer. Using the button **Test connection** we can also test that GeoServer can communicate to GeoFence. If everything is ok, you'll get this message:
```
Comment (bradh) [inline doc/en/user/source/community/geofence/configuration.rst]: typo: altready -> already

**Comment by NielsCharlier** [comment]
Seems to improve documentation and test coverage quite a bit. I had a quick look over the new test cases and they look clear to me. I see some simplifying refactoring too. I have no objections.

**Comment by simboss** [comment]
Should we merge this one? @NielsCharlier @aaime @etj feedback?

**Comment by aaime** [comment]
@afabiani must collects votes for the proposal on the list first, right now the GSIP only has 3 votes

**Pull request merged**